tests/test_framework/mod.rs

```diff
@@ -466,7 +466,8 @@ pub fn verify_swap_results(
         .iter()
         .zip(org_maker_spend_balances.iter())
         .for_each(|(maker, org_spend_balance)| {
-            let wallet = maker.get_wallet().read().unwrap();
+            let mut wallet = maker.get_wallet().write().unwrap();
+            wallet.sync_no_fail();
```

Comment:
I assume this can happen somewhere else also. And I have been hitting some sync failures while doing some local tests. Can we add this syncing at the top of the verify functions and for all verification functions. We should never be doing verification without syncing.

Also many places in tests we are using the `sync()` where we can use the `sync_no_fail` variant which never fails to sync.

This can be done in a followup PR.